Service contracts MUST NOT rely on the execution context (#3870)
magento/devdocs: guides/v2.2/coding-standards/technical-guidelines.md

6.4.4.12. Any customizations to the domain/business logic MUST be executed on the Service Contracts layer, and so MUST be declared in the `global` area of configuration.

> 6.4.4.13. Service contracts MUST NOT rely on the execution context(application area). Services implementation MUST NOT know about the application state.

Reviewer: ```suggestion
6.4.4.13. Service contracts MUST NOT rely on the execution context (application area). Services implementation MUST NOT know about the application state.
```

Author: done

> 6.4.4.13. Service contracts MUST NOT rely on the execution context (application area). Services implementation MUST NOT know about the application state.

Reviewer: ```suggestion
6.4.4.13. A service contract MUST NOT rely on execution context (application area). The service implementation MUST NOT depend on the application state.
```

> 6.4.4.13. Service contracts MUST NOT rely on the execution context (application area). Services implementation MUST NOT know about the application state.

Reviewer 2: Should it be SHOULD or can it be applied in all cases today?

Author: for all cases, for all service contracts
